src/geometry/euclidean/circumcenter.lean

```diff
@@ -378,31 +378,67 @@ begin
           (λ i, hr i (set.mem_univ _))).symm
 end
 
+/-- If there exists a distance that a point has from all vertices of a
+simplex, the orthogonal projection of that point onto the subspace
+spanned by that simplex is its circumcenter.  Version with distance
+condition using `set.range`. -/
+lemma orthogonal_projection_eq_circumcenter_of_exists_dist_eq {n : ℕ} (s : simplex ℝ P n)
+  {p : P} (hr : ∃ r, ∀ p₁ ∈ set.range s.points, dist p₁ p = r) :
+  orthogonal_projection (affine_span ℝ (set.range s.points)) p = s.circumcenter :=
+begin
+  rw exists_dist_eq_iff_exists_dist_orthogonal_projection_eq (subset_affine_span ℝ _) p at hr,
+  cases hr with r hr,
+  rw set.forall_range_iff at hr,
+  exact s.eq_circumcenter_of_dist_eq
+    (orthogonal_projection_mem
+      ((affine_span_nonempty ℝ _).2 (set.range_nonempty _))
+      (submodule.complete_of_finite_dimensional _)
+      p) hr
+end
+
+/-- If there exists a distance that a point has from all vertices of a
+simplex, the orthogonal projection of that point onto the subspace
+spanned by that simplex is its circumcenter.  Version with distance
+condition being for all indicies in `fin (n + 1)`. -/
+lemma orthogonal_projection_eq_circumcenter_of_exists_dist_eq' {n : ℕ} (s : simplex ℝ P n)
+  {p : P} (hr : ∃ r, ∀ i, dist (s.points i) p = r) :
```

Comment:
It seems irritating to repeat this proof for both of:

* `(hr : ∃ r, ∀ p₁ ∈ set.range s.points, dist p₁ p = r)`
* `(hr : ∃ r, ∀ i, dist (s.points i) p = r)`

Which seems equivalent to writing for some `(f : fin n → P) (q : P → Prop)`

* `(h : ∀ fi ∈ set.range f, q fi)`
* `(hr : ∀ i, q (f i))`

Is one of these more canonical? Should `set.forall_range_iff` be tagged a simp lemma?

Reply:
(asked on zulip here: https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/Should.20.60set.2Eforall_range_iff.60.20be.20taged.20.60simp.60.3F/near/209758088)

Reply:
`simp` is now able to translate the first hypothesis into the second